added missing subcodes and improved error message for missing enum values

zawlazaw opened the pull request:
Java's `Status.SubCode` was out of sync with `include/rocksdb/status.h:SubCode`.

When running out of disc space this led to an `IllegalArgumentException` because of an invalid status code, rather than just returning the corresponding status code without an exception.

I added the missing status codes.

By this, we keep the behaviour of throwing an `IllegalArgumentException` in case of newly added status codes that are defined in C but not in Java.

We could think of an alternative strategy: add in Java another code "UnknownCode" which acts as a catch-all for all those status codes that are not yet mirrored from C to Java. This approach would never throw an exception but simply return a non-OK status-code.

I think the current approach of throwing an Exception in case of a C/Java inconsistency is fine, but if you have some opinion on the alternative strategy, then feel free to comment here.

adamretter commented:
@zawlazaw Excellent, thanks for your contribution. For this to work, you will also need to update `toJavaStatusSubCode` in java/rocksjni/portal.h`

zawlazaw commented:
Thanks for pointing this out, I overlooked that there must be some JNI code that creates the Java `Status`-instances. Now I added the missing enum values to `toJavaStatusSubCode` in `portal.h`.

Further, I removed the entry for `kMaxSubCode`, since its value is 8 rather than 0x7E, and since it is not intended to be a sub-code, but it is only used as a size-indicator for an array-allocation.

And last but not least, since in `portal.h` all unknown values are mapped to 0x7F, I added in Java's `Status` another enum value `Undefined` in order to represent this case in `Code` and `SubCode`. In particular, this implies that future additional enum values in `status.h` that are not yet added to `portal.h` will be represented as `Undefined` error status rather than throwing an exception.

adamretter commented:
@zawlazaw LGTM thanks.

@sagar0 can you merge please?

whipit commented:
Did this change make it to a release?  How can I find out?  Thanks;

adamretter commented:
@sagar0 can you comment on which release this is in please?

sagar0 commented:
This commit went first into 5.9 release. 
```
svemuri@devbig ~/rocksdb (master) $ git branch --contains 57fcdc2
  5.10.fb
  5.11.fb
  5.12.fb
  5.13.fb
  5.9.fb
...
* master
...
```
